### ISSUE TITLE
Time Crisis 4 [BLUS30093] only allows semi-automatic firing with SMG/MG in Arcade Mode when using the Guncon 3

### ISSUE THREAD

**Issue by Dooperino** [RPCS3/rpcs3#13191]
### Quick summary

Using the SMG in Time Crisis 4 with a Guncon 3 only allows for semi-automatic firing, rather than automatic firing on trigger pull as it should. 

### Details

With the framelimiter set to 50fps, the Guncon 3 is mostly useable except when using the SMG and helicopter Machine Gun in Arcade mode, which weirdly only register semi-automatic shots on trigger *release* after a trigger pull. Holding the trigger takes up ammo, but does not fire bullets  at enemies until trigger release. 

This only happens in Arcade Mode (traditional shooting mode), and the issue is oddly absent in the Captain Rush FPS Mode (SMGs fire automatically on trigger pull/hold as they should).  

### Attach a log file

[RPCS3.log](https://github.com/RPCS3/rpcs3/files/10368842/RPCS3.log)

### System configuration

OS: Windows 11 Home Single Language 64-bit (10.0, Build 22000) (22000.co_release.210604-1628)
CPU: Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz (12 CPUs), ~3.2GHz
GPU: GALAX NVIDIA GTX 1070 8GB (Driver version 522.25)
RAM: 16GB RAM

### Other details

With the Guncon 3 attached to PC via USB, RPCS3 crashes when exiting Time Crisis 4 normally.

**Comment by Megamouse** [comment]
Please update your game to the latest version and try again

**Comment by Dooperino** [comment]
> Please update your game to the latest version and try again

Thanks for the reply! I believe I've been on the latest patch of the game (v1.01) at the time of posting this issue. 
![image](https://user-images.githubusercontent.com/34993496/211808452-4af249a0-e983-478b-9e46-7ddac0699326.png)

**Comment by Dooperino** [comment]
Update: Found that using the *Automatic Cannon* as Player 2 in the Helicopter segment early in the Aracade mode works as intended and is fully automatic on trigger hold. The *Machine Gun* that Player 1 uses during said segment remains problematic (semi-automatic on trigger hold). 

The bug seems to occur only with machine guns... weird

**Comment by Dooperino** [comment]
Update 2: As of **0.0.26-14604-52b67b46 Alpha**, RPCS3 no longer crashes when closing a game with Guncon 3 attached. Machine gun glitch still present.

**Comment by jasonmac19** [comment]
For what it's worth I'm encountering the same behaviour when using the new emulated GunCon3 feature. Should I raise a separate issue for this?

**Comment by jamesgalllla** [comment]
I tested out the emulated guncon3 feature with my Gun4IR and I have the same issues. Machine gun doesn't work properly for either Player 1 or Player 2 side. Can't seem to get the gun tracking to work in fullscreen as when I tried calibrating in fullscreen the crosshair used for calibration doesn't appear (You can still get rid of the mouse cursor by using programs like nomousy). Couldn't get it to track without setting the mouse handler to basic as setting it to raw didn't seem to work for whatever reason. The multi-screen battle sections also seem impossible to clear. Tried them multiple times but the screen would never move so you're forced to take damage (same thing happens on the arcade version using the Play! emulator) Not sure if anyone using Sinden or Gun4IR has these same issues or if it's just a me thing.

**Comment by kotase2005** [comment]
Today I face to the same thing. Real Guncon 3 on batocera v40 and rpcs3: machine gun does not hit any enemies when holding trigger. I tried to limit fps to 50; 55; 59,94; 60 but the result is the same.

**Comment by Connorsdad** [comment]
Same issue in time crisis 4, the machine gun not registering when holding the trigger.

**Comment by jamesgalllla** [comment]
Really hope someone fixes this... love this game but can't stand to play the move version with the annoying crosshair

**Comment by digant73** [comment]
@Dooperino, @kotase2005, @Connorsdad is that still an issue?

**Comment by TipTopCream** [comment]
Yes, this is still an issue in this version of the game (standalone) no machine gun and Unable to move left or right in the multiscreen parts of the game. 

In the combo version of the game (razing storm) all works well but you are stuck with the crosshairs on!! It just takes away from the game! We need a cheat/patch to remove the crosshair.

**Comment by Bag-Head** [comment]
Wanted to confirm this issue is still present when emulating GunCon3 (for standalone TC4 and the Razing Storm version), though in my case shots for the machine gun only come out when releasing the trigger for both characters. This issue is bad enough that doing the "Terror Bytes" sections is likely impossible without taking damage unless maybe using a turbo function?

**Comment by TipTopCream** [comment]
Bag-head - The stand alone version is just a mess. The machine gun doesn’t work, I tried an AHK script for auto fire.  It’s ok but still not great but further on in the multiscreen parts you cannot change screen. With the 2 glitches it’s just not worth playing. 

TC4 razing storm - if you set your gun to be a motion controller instead of guncon3 the game works 100%!! I have 2 player working with 2 Sinden guns. The only down side, this version of the game has a built-in crosshair and it cannot be removed. 

I have tried unpacking the game files and remove or replace the crosshair with a transparent graphic but it is beyond me. 

Someone out there must be able to do it.

**Comment by Bag-Head** [comment]
> Bag-head - The stand alone version is just a mess. The machine gun doesn’t work, I tried an AHK script for auto fire. It’s ok but still not great but further on in the multiscreen parts you cannot change screen. With the 2 glitches it’s just not worth playing.

By multiscreen parts do you mean where you're behind those barracades and you swap left and right? Because if you're using a Sinden (or emulating Guncon3 in general) you just need to map the dpad to the right analog stick in the pad settings, as that maps to the Guncon3's B stick which lets you swap screens

> TC4 razing storm - if you set your gun to be a motion controller instead of guncon3 the game works 100%!! I have 2 player working with 2 Sinden guns. The only down side, this version of the game has a built-in crosshair and it cannot be removed.

I've messed with Razing Storm and that worked fine for me in Guncon3 emulated mode with sinden, but I've not tried two at a time

> I have tried unpacking the game files and remove or replace the crosshair with a transparent graphic but it is beyond me.
> 
> Someone out there must be able to do it.

For sure someone with interest will be able to one day

**Comment by FANS84** [comment]
Hello. I've played and finished the Complete Mission of standalone version of Time Crisis 4. The SMG problem, doesn't occurs in the FPS chapters. But in the traditional shooting chapters, the SMG remains bugged, since they are the same of arcade mode.  The Terror Bytes sessions were tough for this reason, I could succeed using the shotgun. 

I played using a WiiMote+nunchuk, mounted in a WiiZapper, using the software Gunmote.
